### PR TITLE
Hostile mobs will properly lose targets that leave for other zlevels

### PR DESCRIPTION
**Pull request by ChangelingRain**

### CODE REVIEW

**Comment by optimumtact** [comment]
Remie says this is a dodgy bandaid and that I shouldn't have merged it.

**Comment by RemieRichards** [comment]
I need info on what the hostile mob was chasing.
Given that when in non-object mode a hostile mob finds things by it's ```hearers()```, it should be missing those that are on a different Z, I bring this up as your fix is under and if that requires it to be in the list of valid targets, which if it's on a different Z, it can't be, the only way would be if it was in non-object mode and was either a turret or a mecha

Now, I blame ```can_see()``` one of our ancient procs, and ```get_dist()``` a byond proc that doesn't count Z levels (actually I think it counts it as "1" in the up direction)

Now if this WAS a mecha/turret, thats a little better as the fix actually "works" but it still isn't the right fix. If it wasn't a mecha/turret (as I assume, for 1 turrets can't move on their own (yet, you can mount them to vehicles via badminnery/code) and mecha crossing Zs is also kind of rare) then it will NOT be in the possible targets list, which means it skips the first if AND your fix, eventually falling down to the bottom of the proc, where there's already a ```LoseTarget()```, which is probably why if you tested this (you did test this, yes?) you thought your fix worked.

So to recap:
* Either your fix doesn't work at all and this wasn't something that needed fixing
OR
* It's relying on dodgy (sometimes in-built) assumptions of everything being on the same Z level.
